server/index.js

```diff
@@ -119,14 +119,15 @@ export default class Server {
     }
   }
 
-  async start (port) {
+  async start (port, hostname) {
     await this.prepare()
     this.http = http.createServer(this.getRequestHandler())
     await new Promise((resolve, reject) => {
       // This code catches EADDRINUSE error if the port is already in use
       this.http.on('error', reject)
       this.http.on('listening', () => resolve())
-      this.http.listen(port)
+      if (hostname) this.http.listen(port, hostname)
```

Comment:
taken from my other comment:
> @jessehattabaugh great though actually. Only now I'm remembering that when you pass `-H`  without a hostname it will actually turn into `true`

If you check for `hostname` it will evaluate to true when passing `-H` 😅 It think it's best to check if it is a valid number or something like that 🤔 

cc @nkzawa